.github/workflows/build.yaml

```diff
@@ -0,0 +1,52 @@
+name: Build
+
+on:
+  push:
+    branches:
+      - main
+      - release/*
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: "build"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK for running Gradle
+        uses: actions/setup-java@v3
+        with:
+          distribution: temurin
+          java-version: 17.0.6
+        # TODO: Build and publish a snapshot instead!
+      - name: run gradle
+        run: ./gradlew check
+
+  issue:
+    name: Open issue on failure
+    needs: [ build ]
+    runs-on: ubuntu-20.04
+    permissions:
+      issues: write
+    if: always()
+    steps:
+      # run this action to get workflow conclusion
+      # You can get conclusion by env (env.WORKFLOW_CONCLUSION)
+      - uses: technote-space/workflow-conclusion-action@v3.0.3
```

Comment:
have you seen https://github.com/open-telemetry/opentelemetry-java/blob/main/.github/workflows/reusable-workflow-notification.yml? (we use a copy of this across the other java repos)

Reply:
Oooh, I hadn't seen that. I'll copy it over for consistency then. Thank you!